Hddm python mods

rjones30 opened the pull request:
- program/Utilities/hddm/hddm-py.cpp [rtj]
  - new hddm tool that generates python modules for supporting a
    new python API for reading and writing hddm files and accessing
    their contents. It functions the same way as the existing tools
    hddm-c and hddm-cpp, in that it generates source code that can
    be built into a library and used as a python module. Example:
      $ hddm-py event.xml
    will generate two new source files setup_hddm_X.py and hddm_X.cpy,
    where X is the hddm "class" abbreviation that is defined in the
    HDDM top element within the event.xml input file. Executing the
    python script setup_hddm_X.py with "build", as in the example:
      $ python setup_hddm_X.py build
    will then generate a shared library hddm_X.<suffix> where the
    <suffix> depends on the python version and ends with ".so".
    This .so file can be copied somewhere into your PYTHONPATH and
    used as "import hddm_X" in your python scripts.
- SConstruct, SBMS/sbms.py, libraries/HDDM/SConscript [rtj]
  - add scons support for building python modules and installing
    them in the platform-dependent binaries tree in sim-recon.
  - add specific instructions under the libraries/HDDM part of the
    sim-recon tree to create python hddm modules for streams based
    on event.xml (hddm_s streams), rest.xml (hddm_r streams), and
    mc.xml (hddm_mc_s streams).
  - include support for both python 2.X and 3.X extention modules.
    The same code and build framework should work for both, but
    to avoid possible collisions, the shared objects are installed
    in separate directories within the sim-recon install area.
  - the build procedures assume that the python distutils package
    is installed. This should be default on most systems, but I
    state it here, that these scons builds will fail without the
    python distutils package being installed.

rjones30 committed:
* program/Utilities/hddm/hddm-py.cpp [rtj]
   - new hddm tool that generates python modules for supporting a
     new python API for reading and writing hddm files and accessing
     their contents. It functions the same way as the existing tools
     hddm-c and hddm-cpp, in that it generates source code that can
     be built into a library and used as a python module. Example:
        $ hddm-py event.xml
     will generate two new source files setup_hddm_X.py and hddm_X.cpy,
     where X is the hddm "class" abbreviation that is defined in the
     HDDM top element within the event.xml input file. Executing the
     python script setup_hddm_X.py with "build", as in the example:
        $ python setup_hddm_X.py build
     will then generate a shared library hddm_X.<suffix> where the
     <suffix> depends on the python version and ends with ".so".
     This .so file can be copied somewhere into you PYTHONPATH and
     used as "import hddm_X" in your python scripts.

* SConstruct, SBMS/sbms.py, libraries/HDDM/SConscript [rtj]
   - add scons support for building python modules and installing
     them in the platform-dependent binaries tree in sim-recon.
   - add specific instructions under the librarys/HDDM part of the
     sim-recon tree to create python hddm modules for streams based
     on event.xml (hddm_s streams), rest.xml (hddm_r streams), and
     mc.xml (hddm_mc_s streams).
   - include support for both python 2.X and 3.X extention modules.
     The same code and build framework should work for both, but
     to avoid possible collisions, the shared objects are installed
     in separate directories within the sim-recon install area.
   - the build procedures assume that the python distutils package
     is installed. This should be default on most systems, but I
     state it here, that these scons builds will fail without the
     python distutils package being installed.

rjones30 committed:
* programs/Utilities/hddm/SConscript [rtj]
   - add an explicit dependency that ensures that the hddm-py tool is built
     before any of the builders in libraries/HDDM try to use it to build
     python modules for any of the hddm classes.

gluex commented:
Build status for this pull request: FAILURE

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_python_mods

rjones30 committed:
* SBMS/sbms.py [rtj]
   - customize the name of the temporary build directory so that
     multiple simultaneous scons threads building different python
     extension modules do not collide.
* programs/Utilities/hddm/hddm-py.cpp [rtj]
   - changes to the setup_hddm_X.py build script to support the
     above changes to sbms.py.

gluex commented:
Test status for this pull request: SUCCESS

  Summary: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/tests/log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/tests-logs)

 Build log: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/make_hddm_python_mods.log)
 Build report: [/work/halld/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt](https://halldweb.jlab.org/pull_request_test/sim-recon^hddm_python_mods/report_hddm_python_mods.txt)
 Location of build: /work/halld/pull_request_test/sim-recon^hddm_python_mods
